test/jdk/javax/swing/JFileChooser/HTMLFileName.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.Icon;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.filechooser.FileSystemView;
+import java.io.File;
+import java.util.List;
+
+/*
+ * @test
+ * @bug 8139228
+ * @summary JFileChooser should not render Directory names in HTML format
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual HTMLFileName
+ */
+
+public class HTMLFileName {
+
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                1. FileChooser shows up a virtual directory and file with name
+                   "<html><h1 color=#ff00ff><font face="Comic Sans MS">Testing Name".
+                2. On "HTML disabled" frame :
+                      a. Verify that the folder and file name must be plain text.
+                      b. If the name in file pane window and also in directory
+                         ComboBox remains in plain text, then test PASS.
+                         If it appears to be in HTML format with Pink color,
+                         then test FAILS.
```

Comment:
```suggestion
                         ComboBox remains in plain text, then test passes.
                         If it appears to be in HTML format with Pink color,
                         then test fails.
```
You could be more specific: _“click **Fail**”_ if a condition isn't met. (Otherwise, continue with the test.) End the instructions with _“click **Pass**”_.

Anyway, the instructions are pretty clear as they are, so real change is required.

Reply:
You may want to switch to HTML-formatted instructions to alleviate differences in the default fonts in different L&Fs.

Reply:
Why would we need HTML-formatted instructions btw ?

Reply:
> Why would we need HTML-formatted instructions btw ?

Because the instructions are really hard to read in Windows L&F (because of the monospace font):

![HTMLFileName Instructions in Windows L&F: the text doesn't fit nicely in the window and gets wrapped](https://github.com/user-attachments/assets/ecd39482-e448-4436-96e5-8d6b96c888d9)

The image isn't clear, but you get the idea: all the text gets wrapped, so it's hard to keep track of the instructions. Anyway, I'm fine with it.

Harshitha had [a similar problem](https://github.com/openjdk/jdk/pull/24456#discussion_r2054468745), and she switched to [using HTML formatting](https://github.com/openjdk/jdk/pull/24456#discussion_r2054516242) which works reliably across different OS.

You can check [`JFileChooser/bug4943900.java`](https://github.com/openjdk/jdk/blob/master/test/jdk/javax/swing/JFileChooser/bug4943900.java) on how it looks and works in different OS.

https://github.com/openjdk/jdk/blob/81536830ed096005c4f09ab446238ce50989cea9/test/jdk/javax/swing/JFileChooser/bug4943900.java#L40-L47

Reply:
Yup got it. I've update the test instructions to HTML-formatted.